components/automate-ui/src/app/helpers/auth/regex.ts

```diff
@@ -20,7 +20,13 @@ export class Regex {
     NO_MIXED_WILDCARD_ALLOW_HYPHEN: '^(\\*|[-\\w]+)$',
 
     // Allows all special characters except colon :
-    NO_MIXED_WILDCARD_ALLOW_SPECIAL: '^(\\*|[^:*]+)$'
+    NO_MIXED_WILDCARD_ALLOW_SPECIAL: '^(\\*|[^:*]+)$',
+
+    // Allows valid FQDN only
+    VALID_FQDN: '(https?://)?([\\da-z.-]+)\\.([a-z.]{2,6})[/\\w .-]*/?',
+
+    // Allows valid IP Address only (ipv4)
+    VALID_IP_ADDRESS: '^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$'
```

Comment:
If you copied these from a published source it is important to add a comment noting the source.
If you created these from scratch, I think it would be really good to add tests (at least a couple dozen for each) that show what does and does not validate. That will let folks who are less familiar with reading regex's be able to judge whether it meets product needs.

Just a few problem examples off-hand for FQDN:

These should match but do not:
```
    http://fo_o.bar/
    https://bit.ly/2OWGwiL
```
These should not match but do:
```
    http://foo.bar com/
    http://foo.bar..com/
    http://...foo.bar.com/
    http://foo.bar-.-.
```

Probably better to find a reliable, documented source rather than create your own to avoid such problem areas. 😁

Reply:
Hello @msorens 
I have rewrote the ```REGX``` for FQDN but still not able to create a perfect one so can I get a help on the same please.

Reply:
I did a bit of web searching and found a couple possibilities but because of the nature of an FQDN there does not seem to be a consensus of a good matching regex. So my suggestion is let's back up; you should chat with whoever is requesting this work (is that @jonong1972 ?) to find out what types of FQDN's are needed. It may be that for chef servers we can more precisely specify a URL.

Reply:
I found that we use this in a couple spots in the code base:
https://github.com/chef/automate/blob/f8ca770fc9c7915a9658855d5130811e5875eae1/components/automate-ui/src/app/pages/notification-form/notification-form.component.html#L97
https://github.com/chef/automate/blob/f8ca770fc9c7915a9658855d5130811e5875eae1/components/automate-ui/src/app/pages/data-feed-form/data-feed-form.component.html#L37
which, in English means: _either `http://` or `https://` followed by something-dot-something._
It is a very forgiving pattern--it will not exclude any valid ones, which is important--though it will certainly allow invalid ones.

@jonong1972 Do you think that pattern would be sufficient here?

Reply:
@msorens created followed regx which is allowing most of valid ones and blocking most of invalid one 

`VALID_FQDN: /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-zA-Z0-9_]+([\-\.]{1}[a-zA-Z0-9]+)*\.[a-zA-Z]{2,5}(:[0-9]{1,5})?(\/.*)?$/,`

tried and added test cases for the examples which you have provided 
Valid Inputs
```
http://fo_o.bar/
https://bit.ly/2OWGwiL
```

Invalid Inputs

```
 http://foo.bar com/
 http://foo.bar..com/
 http://...foo.bar.com/
 http://foo.bar-.-.
```

working well with above all can we go with this ?

Reply:
I want to get some product or UX guidance on this question--I will try to speak to @jonong1972 about this tomorrow.

Reply:
@sachin-msys lets not do any validation on this input right now, we can come back and iterate later after this work is out of discovery.

Reply:
@susanev yes sure sounds good, for now resolved conflict just now please have a look thanks